src/server/models/Reading.js

```diff
@@ -475,6 +475,55 @@ class Reading {
 		return groupThreeDData;
 	}
 
+	/**
+	 * Gets the data range (min and max timestamps) for a single meter
+	 * Used for 3D auto-adjustment to determine available data range
+	 * @param meterID The meter ID to get data range for
+	 * @param conn the connection to use
+	 * @returns {Promise<{minDate: Moment | null, maxDate: Moment | null}>}
+	 */
+	static async getMeterDataRange(meterID, conn) {
+		const result = await conn.oneOrNone(`
+			SELECT 
+				MIN(start_timestamp) as min_date,
+				MAX(end_timestamp) as max_date
+			FROM readings 
+			WHERE meter_id = $1
+		`, [meterID]);
+		if (result && result.min_date && result.max_date) {
```

Comment:
My testing found (as I would expect) that if there are not readings so no min/max then you get:

result:  { min_date: null, max_date: null }

Thus, I don't think you need the extra logic for this.